autograd.h

```diff
@@ -36,10 +43,39 @@ using Tensor = tag::Tensor;
 using Container = std::shared_ptr<ContainerImpl>;
 using Optimizer = std::shared_ptr<OptimizerImpl>;
 
-void backward(Variable loss, bool keep_graph=false);
 void backward(Tensor loss, bool keep_graph=false);
-void save(std::string fn, Container const model);
-void load(std::string fn, Container model);
+
+// Some convenience functions for saving and loading
+template <typename T>
+void save(std::ostream& stream, T const & obj) {
+  cereal::BinaryOutputArchive archive(stream);
+  archive(*obj);
+}
+template <typename T>
+void load(std::istream& stream, T& obj) {
+  cereal::BinaryInputArchive archive(stream);
+  archive(*obj);
+}
+template <typename T>
+void save(std::ostream& stream, T const * obj) {
+  cereal::BinaryOutputArchive archive(stream);
+  archive(*obj);
+}
+template <typename T>
+void load(std::istream& stream, T* obj) {
+  cereal::BinaryInputArchive archive(stream);
+  archive(*obj);
+}
+template <typename T>
+void save(std::string const& fn, T const & obj) {
```

Comment:
Nit: `path` or `fileName` would be a bit more expressive and tell you the purpose on first glance.